### PR TITLE
ESQL: Remove OPTIONS clause in FROM command (#108692)

### PR DESCRIPTION
**Pull request by bpintea**
This remove the `OPTIONS` clause of the `FROM` command.

(cherry picked from commit 3b21ec88180e0d1597c3fe84e956634cbad37681)

### CODE REVIEW

**Comment by bpintea** [comment]
@elasticmachine update branch

**Comment by bpintea** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java [hunk 1]
```diff
@@ -797,24 +797,47 @@ static EsRelation readEsRelation(PlanStreamInput in) throws IOException {
...
-            relation.esSourceOptions().writeEsSourceOptions(out);
+        if (supportingEsSourceOptions(out.getTransportVersion())) {
+            writeEsSourceOptions(out); // write (null) string fillers expected by remote
         }
         out.writeBoolean(relation.frozen());
     }
 
+    private static boolean supportingEsSourceOptions(TransportVersion version) {
+        return version.onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)
+            && version.before(TransportVersions.ESQL_REMOVE_ES_SOURCE_OPTIONS_8_14);
```
Comment (dnhatn) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java:820]: BWC tests between 8.15 and 8.14 will fail with the new patch version. The new patch version 8_636_00_2 doesn't expect esSource, but 8.15 will send esSource for this patch version.

Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java]: Thanks Nhat, went with removing the version upper bound check, as you suggested. 🙏

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @bpintea.

**Pull request merged**